drivers/i3c/i3c_dw.c

```diff
@@ -2385,11 +2427,27 @@ static int dw_i3c_init(const struct device *dev)
 	}
 #endif /* CONFIG_I3C_CONTROLLER */
 	dw_i3c_enable_controller(config, true);
+	enable_interrupts(dev);
+
+#ifdef CONFIG_I3C_TARGET
+	ret = dw_i3c_init_scl_timing(dev, ctrl_config);
+	if (ret != 0) {
+		LOG_ERR("Normal Speed set failed");
+		return ret;
+	}
+#endif /* CONFIG_I3C_TARGET */
 #ifdef CONFIG_I3C_CONTROLLER
+
 	if (!(ctrl_config->is_secondary)) {
 		/* Perform bus initialization - skip if no I3C devices are known. */
 		if (config->common.dev_list.num_i3c > 0) {
 			ret = i3c_bus_init(dev, &config->common.dev_list);
+		} else {
+			ret = dw_i3c_init_scl_timing(dev, ctrl_config);
+			if (ret != 0) {
+				LOG_ERR("Normal Speed set failed");
+				return ret;
+			}
```

Comment:
This needs to be reverted back to what it is above as this won't work as it's not calling scl timing..

Reply:
> The timing config happens twice when the device is a controller if i keep it general as before. Once by default and one more time at i3c_bus_init() function.

yes... but that's okay for it be called twice, my concern is that `i3c_bus_init()` is a public and can be called easily outside. It's also possible that `i3c_bus_init` could change in future, I'd rather see this be defensive and program it's own scl timing before it calls that `i3c_bus_init`.